### DISCUSSION TITLE
World Package 1.3.0버전이 업데이트 되었습니다

### DISCUSSION THREAD

**Discussion by Zepeto-TechEvangelist** [naverz/zepeto-studio-kor#677] [1. Announcements]
안녕하세요.
World Package 1.3.0버전이 업데이트 되어서 안내드립니다.

### 안내
[패키지 업데이트 방법](https://studio.zepeto.me/kr/guides/updating-package)
[1.3.0 릴리즈 노트 보기](https://studio.zepeto.me/kr/guides/world-130)

<hr/>

### 신규 추가 기능

1. CharacterController - CreateByUserId / CreateModelByZepetoID / CreateModelByUser API 추가
- 가이드 보기 : [2-1. ZepetoCharacter](https://studio.zepeto.me/guides/zepetocharacter)

2. CharacterController - Motion V1/V2 캐릭터 동작에 대한 Walk / Run Threahold 변경 옵션 추가
- 가이드 보기 : [2. Charactor Control](https://studio.zepeto.me/guides/character-control)

3. 동영상 촬영 기능 API 제공
- 관련 이슈 : https://github.com/naverz/zepeto-studio-kor/discussions/656
- 가이드 보기 : [9-2. 동영상 저장/공유/피드 올리기](https://studio.zepeto.me/guides/video-save-share-post)

4. 퀵채팅 - GetQuickMessageList / SendQuickMessage API 추가
- 가이드 보기 : [9-3. 퀵채팅 사용하기](https://studio.zepeto.me/guides/quick-message)

5. Unity.Timeline 사용 지원
- 관련 이슈 : https://github.com/naverz/zepeto-studio-kor/discussions/654
- https://docs.unity3d.com/kr/2018.4/Manual/TimelineOverview.html

6. OnPreCull, OnWillRenderObject 이벤트 함수 추가
- 관련 이슈 :  https://github.com/naverz/zepeto-studio-kor/discussions/606
- https://docs.unity3d.com/kr/530/ScriptReference/Camera.OnPreCull.html
- https://docs.unity3d.com/kr/530/ScriptReference/MonoBehaviour.OnWillRenderObject.html

<hr/>

### 개선된 기능
-  CharacterController - Move 중 Teleport 시, 간헐적으로 Teleport 동작이 무시되는 이슈 수정 https://github.com/naverz/zepeto-studio-kor/discussions/556
-  CharacterController - ZepetoScreenStick, ZepetoScreenTouchpad 컴포넌트 OnDragEvent 바인딩 이슈 수정 https://github.com/naverz/zepeto-studio-kor/discussions/593
-  UnityEngine.InputSystem - InputAction에 대한 이벤트 바인딩 시, CallbackContext 읽어오지 못하는 이슈 수정 https://github.com/naverz/zepeto-studio-kor/discussions/223
-  리더보드 추가 시에 발생하는 로딩 문제 수정 https://github.com/naverz/zepeto-studio-kor/discussions/424
-  IWP 결제 관련 오류 수정 https://github.com/naverz/zepeto-studio-kor/discussions/558
-  테스트 접속 링크로 월드 접속시 iOS에서 초대가 안되는 문제 수정 https://github.com/naverz/zepeto-studio-kor/discussions/567
-  서버 코드에 중괄호를 넣지 않으면 패키지 업로드가 정상적으로 되지 않는 문제 수정
- Character Controller Motion V2의 Gravity 값을 에디터 인스펙터에서 노출되지 않게 변경
   - character.MotionV2.Gravity = (number); 을 통한 접근은 기존과 동일하게 가능합니다.

**Comment by smilejsu82** [comment]
wow 😃

**Comment by Andwhy81kr** [comment]
1.3.0 패키지 업데이트 후 타입스크립트가 빌드가 되지 않고 있습니다.
혹시 이슈가 있나요?

오류창 스샷입니다.
![20220428_231208_2](https://user-images.githubusercontent.com/45650836/165772137-2bc4a4f7-e051-4de3-9388-c10b37c5cba3.png)
![20220428_231208_1](https://user-images.githubusercontent.com/45650836/165772152-5f07dfe3-8e43-4123-a830-bd4588a7357a.png)?

**Reply by Andwhy81kr** [reply]
관련있는 문제인지 모르겠지만, 1.3.0버전에선 ZepetoWorldMultiplay 가 사라지고, ZepetoDynamicMultiplay, ZepetoStanrdMultiplay 등으로 교체된것 같은데, 이건어떻게 설정해야하나요?
요것땜에 스크립트가 Missing나는경우가 있는듯합니다.

**Reply by Zepeto-TechEvangelist** [reply]
안녕하세요, 
https://github.com/naverz/zepeto-studio-kor/discussions/660
위 이슈와 동일한 현상으로 작업중인 프로젝트에서 월드 버전을 업데이트 할 경우 Assets > Reimport All을 수행하면 정상적으로 나타납니다. 
감사합니다.

**Reply by Andwhy81kr** [reply]
위의 타입스크립트 빌드 실패때문인지 ReimportAll 자체가 실패하여 ZepetoWorldMultiplay 도 사용불가하네요.

**Reply by Zepeto-TechEvangelist** [reply]
스크립트를 우선 remove 하신 후에  ReimportAll을 시도해 보시겠어요?

**Reply by Andwhy81kr** [reply]
해당스크립트를 제거하고 빌드해도 동일합니다.
(스크립트 디펜던시가 있어서 다시 작성해서 빌드해도 동일하네요.)

**Reply by Andwhy81kr** [reply]
할수있는 방법을 모두 시도해봐도 해결이 안되네요.
새프로젝트라면 모르겠지만, 현재 프로젝트의 SDK  업데이트는 포기하겠습니다.

**Reply by Andwhy81kr** [reply]
마지막 방법으로 sdk1.1.3버전부터 릴리즈 노트에 있는 SDK버전을 한단계씩 높여가며 테스트중입니다.
SDK 1.1.6버전부터 동일한 에러가 발생되고 있네요.

**Comment by ted002** [comment]
와! 영상기능! 감사합니다!

**Comment by mahoworld** [comment]
혹시 빌드잇 로더 버전 업데이트...계획도 있으실까요? ㅠ0ㅠ

**Reply by Zepeto-TechEvangelist** [reply]
안녕하세요, 빌드잇 로더 계획은 중장기적으로 바라보고 있습니다.
당장 사용하기 불편한 이슈가 있다면 이슈 게시판을 통해서 제보 부탁 드립니다.
감사합니다.

**Comment by Andwhy81kr** [comment]
SDK 1.3.0 버전으로 테스트 결과,

**IWP 결제 관련 오류 수정 https://github.com/naverz/zepeto-studio-kor/discussions/558**

수정되지 않고 여전히 발생하고 있습니다.

확인부탁드립니다.
